src/core/server/saved_objects/serialization/serializer.ts

```diff
@@ -5,26 +5,27 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
+
 import typeDetect from 'type-detect';
-import { LEGACY_URL_ALIAS_TYPE } from '../object_types';
-import { decodeVersion, encodeVersion } from '../version';
-import { ISavedObjectTypeRegistry } from '../saved_objects_type_registry';
-import { SavedObjectsUtils } from '../service';
-import {
+import type {
+  ISavedObjectTypeRegistry,
+  ISavedObjectsSerializer,
   SavedObjectsRawDoc,
   SavedObjectSanitizedDoc,
   SavedObjectsRawDocParseOptions,
-} from './types';
+} from '@kbn/core-saved-objects-server';
+import { LEGACY_URL_ALIAS_TYPE } from '../object_types';
+import { decodeVersion, encodeVersion } from '../version';
+import { SavedObjectsUtils } from '../service';
 
 /**
- * A serializer that can be used to manually convert {@link SavedObjectsRawDoc | raw} or
- * {@link SavedObjectSanitizedDoc | sanitized} documents to the other kind.
+ * Core internal implementation of {@link ISavedObjectsSerializer}
  *
  * @remarks Serializer instances should only be created and accessed by calling {@link SavedObjectsServiceStart.createSerializer}
  *
- * @public
+ * @internal
  */
-export class SavedObjectsSerializer {
+export class SavedObjectsSerializer implements ISavedObjectsSerializer {
```

Comment:
Now that we have a proper interface around the class, it was be switched to internal.

Note that the class is still exported from the entrypoint because we have one (1!) usage of the concrete implementation in a plugin's test file.

This can be addressed later by the introduction of a mock (that we currently don't have for the serializer, surprisingly)

Reply:
> This can be addressed later by the introduction of a mock (that we currently don't have for the serializer, surprisingly)
interesting indeed.